### PR TITLE
Test Coverage PR-3: Tests added in ./cmd subdir

### PR DESCRIPTION
**Pull request by mohanmanikanta2299**
## Summary
This PR is the 3rd part of the test coverage PRs intended to increase the test coverage of the entire repo from 55% to 80+%, fulfilling our coverage requirements. 

## Changes Made
* Implemented comprehensive table-driven tests for the `cmd` package.
* `debug.go`, `dispatch.go`, `init.go`, `report.go`, `root.go`, `utils.go` -- list of files in `cmd` package for which the test have been created/updated.
* Added validation for CLI command flags, ensuring default values and flag bindings are properly configured.
* Added strict assertions (`assert.Equal`, `assert.Contains`) and explicit error checking to validate function outputs and edge cases.
* Used `require` for critical initializations (like loggers and flag lookups) to prevent nil-pointer panics during test execution.

## Testing
* Ran `go test -v ./...` (All tests passing)
* Ran `go test -race ./...` (No race conditions detected)
* Verified coverage metrics locally.

### :link: External Links
https://hashicorp.atlassian.net/browse/CCEN-512

### :thinking: Can be merged upon approval?
:white_check_mark:

## PCI review checklist

- [ ] I have documented a clear reason for, and description of, the change I am making.

- [ ] If applicable, I've documented a plan to revert these changes if they require more than reverting the pull request.

- [ ] If applicable, I've documented the impact of any changes to security controls.

  Examples of changes to security controls include using new access control methods, adding or removing logging pipelines, etc.

### CODE REVIEW

File: cmd/debug_test.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	rootCmd.SetArgs([]string{"debug", "--dirPath", "."})
+
+	err := rootCmd.Execute()
+	// This may fail if no git repo is detected, but should not panic
+	if err == nil {
+		output := buf.String()
+		assert.Contains(t, output, "Copywrite Version:")
+		assert.Contains(t, output, "Working Directory:")
+	}
+}
```
[the remark this answers is not in the document]
Comment (mohanmanikanta2299) [reply cmd/debug_test.go:67]: Refactored Test Case.

File: cmd/root_test.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	assert.Error(t, err)
+}
+
+func TestRootCmd_ConfigFlag(t *testing.T) {
+	flag := rootCmd.PersistentFlags().Lookup("config")
+	require.NotNil(t, flag)
+	assert.Equal(t, ".copywrite.hcl", flag.DefValue)
+}
+
+func Test_initConfig_FileNotExist(t *testing.T) {
```
Comment (sonamtenzin2) [inline cmd/root_test.go:63]: What is this test checking for ?

File: cmd/root_test.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	oldCfgPath := cfgPath
+	defer func() { cfgPath = oldCfgPath }()
+
+	cfgPath = "/tmp/nonexistent_copywrite_test_config_12345.hcl"
```
Comment (sonamtenzin2) [inline cmd/root_test.go]: The `tmp` path is not present for windows systems. Try using the line `filepath.Join(t.TempDir(), "nonexistent.hcl") ` to keep it agnostic of the underlying running system

File: cmd/init_test.go [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+	assert.Contains(t, output, "header_ignore")
+	assert.Contains(t, output, "# (OPTIONAL)")
+}
+
+func TestInitCmd_Run_NoTTY(t *testing.T) {
+	tmpDir := newGitRepo(t, time.Now())
+	t.Chdir(tmpDir)
+
+	buf := new(bytes.Buffer)
+	restoreRootCmd(t)
```
Comment (sonamtenzin2) [inline cmd/init_test.go]: It is usually more idiomatic to have the cleanup commands run after a test is done, so that other tests don't have to rely on doing a cleanup beforehand.

File: cmd/init_test.go [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+			assert.Equal(t, tt.defValue, flag.DefValue)
+		})
+	}
+}
+
+func TestInitCmd_Help(t *testing.T) {
+	buf := new(bytes.Buffer)
+	initCmd.SetOut(buf)
+	initCmd.SetErr(buf)
+	initCmd.SetArgs([]string{"--help"})
```
Comment (sonamtenzin2) [inline cmd/init_test.go:174]: Please check that cleanup is done for all the commands to prevent non-deterministic between runs.

File: cmd/utils_test.go [hunk 6]
```diff
@@ -52,3 +65,112 @@ func Test_colorize(t *testing.T) {
...
+			version:  "1.2.3",
+			commit:   "abc1234",
+			expected: "1.2.3-abc1234",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			oldVersion := version
+			oldCommit := commit
+			defer func() {
```
Comment (sonamtenzin2) [inline cmd/utils_test.go]: t.Cleanup can be used here for testcases

Comment (mohanmanikanta2299) [reply cmd/utils_test.go]: Added cleanup instead of defer func()

File: cmd/init_test.go (hunk 5 above)
Comment (mohanmanikanta2299) [reply cmd/init_test.go:174]: Created restoreInitCmd in helpers and used it for cleanup

File: cmd/init_test.go (hunk 4 above)
Comment (mohanmanikanta2299) [reply cmd/init_test.go]: Used helper functions that perform cleanup at the end of the test case and restore original cmd State

File: cmd/root_test.go (hunk 3 above)
Comment (mohanmanikanta2299) [reply cmd/root_test.go]: Added filepath.Join instead of `/tmp/nonexistent_copywrite_test_config_12345.hcl` in the test case for supporting Windows system.

File: cmd/root_test.go (hunk 2 above)
Comment (mohanmanikanta2299) [reply cmd/root_test.go:63]: This test verifies that initConfig() is a no operation when the config file doesn't exist, it should not panic or modify the global conf state. The assertions confirm that conf still holds its default values (empty License, SchemaVersion = 1) after the call, proving no accidental mutation occurred.

File: licensecheck/copyright_test.go [hunk 7]
```diff
@@ -264,12 +267,108 @@ func TestHasCopyright(t *testing.T) {
...
+func TestHasMatchingCopyright_ErrorHandling(t *testing.T) {
+	t.Run("error on non-existent file", func(t *testing.T) {
+		hasCopyright, err := HasMatchingCopyright("/nonexistent/file.txt", "Copyright", false)
+		assert.NotNil(t, err)
+		assert.False(t, hasCopyright)
+	})
+}
+
+func TestHasMatchingCopyright_EdgeCases(t *testing.T) {
+	AppFs := afero.NewOsFs()
```
Comment (sonamtenzin2) [inline licensecheck/copyright_test.go:291]: Why do we need this, when temp files can be created by something like this ?
`os.CreateTemp(tempDir, "data-*.txt")`

**Review by sonamtenzin2 — APPROVED** [review]

File: licensecheck/copyright_test.go (hunk 7 above)
Comment (sonamtenzin2) [reply licensecheck/copyright_test.go:291]: old changes, keeping as it is for now

Comment (mohanmanikanta2299) [reply licensecheck/copyright_test.go:291]: Will address them in a separate PR.
